src/layouts/sidebar-sidecar/index.tsx

```diff
@@ -17,21 +18,90 @@ const SidebarSidecarLayout = ({
   sidebarProps,
   sidebarSlot,
   sidecarSlot,
+  sidebarPropsLevels,
 }: SidebarSidecarLayoutProps) => {
+  const { isDesktop } = useDeviceSize()
+  const numSidebarLevels = sidebarPropsLevels ? sidebarPropsLevels.length : 0
+  const initialSidebarLevel = useMemo(() => {
+    if (numSidebarLevels > 0) {
+      return numSidebarLevels - 1
+    } else {
+      return 0
+    }
+  }, [numSidebarLevels])
+
+  // If `sidebarPropsLevels` is given, set current level to last level.
+  const [currentSidebarLevel, setCurrentSidebarLevel] =
+    useState<number>(initialSidebarLevel)
+
+  /**
+   * @TODO I think this is temporary until all the pages have
+   * `sidebarPropsLevels`. Otherwise, when we switch from something like
+   * /waypoint to /waypoint/docs, the sidebar level that shows is the first one
+   * because this layout component isn't getting re-rendered in a way that
+   * re-initializes `currentSidebarLevel`.
+   */
+  useEffect(() => {
+    setCurrentSidebarLevel(initialSidebarLevel)
+  }, [initialSidebarLevel])
+
+  /**
+   * Initializes sidebarProps based on the current device size, number of
+   * sidebar props levels given, and current sidebar level.
+   */
+  const finalSidebarProps: SidebarProps = useMemo(() => {
+    // Make backwards compatible with `sidebarProps`
+    if (!sidebarPropsLevels) {
+      return sidebarProps
+    }
+
+    // Copy provided props so we don't modify the original
+    const propsCopy = {
+      ...sidebarPropsLevels[currentSidebarLevel],
+    }
+
+    /**
+     * If desktop, we only care about `backToLinkProps`. Otherwise, tell each
+     * level's Button what to do `onClick`.
+     */
+    if (isDesktop) {
+      delete propsCopy.levelButtonProps
+    } else {
+      delete propsCopy.backToLinkProps
+
+      if (numSidebarLevels > 1) {
+        propsCopy.levelButtonProps.onClick = () => {
+          setCurrentSidebarLevel((prevLevel: number) => {
+            if (prevLevel === 0) {
+              return prevLevel + 1
+            } else {
+              return prevLevel - 1
+            }
+          })
+        }
+      }
+    }
```

Comment:
💭  This feels like something that the sidebar could handle. Instead of deleting props, we can just choose not to render the `levelButton` / `backToLink`. What do you think about that?

Reply:
I think we could invoke `useDeviceSize` in `Sidebar` and let it choose between `backToLinkProps` and `levelButtonProps` based on `isDesktop`, is that along the lines of what you were thinking? I think the `onClick` setting is still needed here, but otherwise I'd be down to not `delete` object properties here! 😁

Reply:
That sounds good to me!